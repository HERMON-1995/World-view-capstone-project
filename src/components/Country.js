import React from 'react';
import PropTypes from 'prop-types';

const Country = (props) => {
  const {
    name, id, population, flag,
  } = props;
  return (
    <>
      <div className="cont" id={id}>
        <div className="img-ico">
          <img src={flag} alt="a country flag" className="flags" />
          <i className="fa-solid fa-circle-arrow-right" />
        </div>
        <div className="text-desc">
          <p className="country-name">{name}</p>
          <p className="population">{population}</p>
        </div>
      </div>
    </>
  );
};

Country.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  flag: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
};

export default Country;
