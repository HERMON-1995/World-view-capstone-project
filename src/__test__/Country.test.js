import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import React from 'react';
import Country from '../components/Country';
import store from '../redux/Store';
import '@testing-library/jest-dom/extend-expect';

window.scrollTo = jest.fn();

describe('Details Component', () => {
  it('should render the details correctly', async () => {
    window.scrollTo.mockClear();
    const component = render(
      <BrowserRouter>
        <Provider store={store}>
          <Country />
        </Provider>
        ,
      </BrowserRouter>,
    );
    expect(component.asFragment()).toMatchSnapshot();
  });
});
