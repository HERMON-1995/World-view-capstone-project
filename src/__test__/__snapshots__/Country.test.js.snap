// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component should render the details correctly 1`] = `
<DocumentFragment>
  <div
    class="cont"
  >
    <div
      class="img-ico"
    >
      <img
        alt="a country flag"
        class="flags"
      />
      <i
        class="fa-solid fa-circle-arrow-right"
      />
    </div>
    <div
      class="text-desc"
    >
      <p
        class="country-name"
      />
      <p
        class="population"
      />
    </div>
  </div>
  ,
</DocumentFragment>
`;
